modules/storages/app/validator/nextcloud_compatible_host_validator.rb

```diff
@@ -31,23 +31,48 @@ class NextcloudCompatibleHostValidator < ActiveModel::EachValidator
   def validate_each(contract, attribute, value)
     return unless contract.model.changed_attributes.include?(attribute)
 
+    validate_capabilities(contract, attribute, value)
+    validate_setup_completeness(contract, attribute, value) if contract.errors.empty?
+  end
+
+  private
+
+  def validate_capabilities(contract, attribute, value)
     response = request_capabilities(value)
-    error_type = check_response(response)
+    error_type = check_capabilities_response(response)
+
     if error_type
       contract.errors.add(attribute, error_type)
       Rails.logger.info(message(value, response, error_type))
```

Comment:
just saw it, is it common, that we log `info` when we have an error occured in some contract?

Reply:
No idea. I simply copied it over from another existing validator. I don't recall any guideline on when to log and when not.

Regarding the log level: I think `info` is right. `:error` would be wrong as the validation works as expected.